### DISCUSSION TITLE
[Core Protocol] is the `l1origin` from RPC `optimism_outputAtBlock` reliable?

### DISCUSSION THREAD

**Discussion by johnhckuo** [ethereum-optimism/developers#514] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I noticed that for some third party providers they provide this RPC method `optimism_outputAtBlock`

curl example:
```
curl --location 'https://sepolia.base.org' \
--header 'Content-Type: application/json' \
--data '{
    "method": "optimism_outputAtBlock",
    "params": [
        "0xb0aec2"
    ],
    "id": 1,
    "jsonrpc": "2.0"
}'
```

its corresponding example is:
```
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "version": "0x0000000000000000000000000000000000000000000000000000000000000000",
        "outputRoot": "0x67c8b6d2f87e5c3146571ae40f5456b99735a8f5da02fd1718a669c85b1277f1",
        "blockRef": {
            "hash": "0xc12766e15c5a3d8df779aa21eeeb605378af7cf6c2eda55a64bdd7302b5401fc",
            "number": 11579074,
            "parentHash": "0x0b3c9846f0a63ce7d0a8a7e51c51abd22b090cc070782c1d3e7b64a0d9c9efdc",
            "timestamp": 1718926436,
            "l1origin": {
                "hash": "0x9acd774c9fc5b21727cb2ee9a4318129470f85c9107d4f0b328a35ae108fa08a",
                "number": 6150841
            },
            "sequenceNumber": 0
        },
        "withdrawalStorageRoot": "0x3fc5c63344a198c5d9f55825b2c255c317fc2fa57783986669d4554abdc83478",
        "stateRoot": "0xbb59eae72a5926541ebda9c2df020ff93b11195f9c69acb2b3bfd1b0de2c1ed1",
        "syncStatus": {
                ...... truncated......
        }
    }
}
```

the value of `result.blockRef.l1origin` if I understand correctly, is the corresponding L1 block number of this L2 block.
Does that means that all L2 transactions can be seen as `safe` after this L1 block? 
If not, how do I derive the correct or the oldest l1 origin block number for an Optimism block?

Thanks!

### Additional Information

This rpc is provided by quicknode: https://www.quicknode.com/docs/optimism/optimism_outputAtBlock
I've replaced the url inside the curl example since there are credentials involved.

**Comment by imtipi** [comment]
> Does that means that all L2 transactions can be seen as safe after this L1 block?

I don't think so,by compare the result.blockRef.l1origin and result.syncStatus.safe_l2.l1origin ,you can see syncStatus's is earlier

> If not, how do I derive the correct or the oldest l1 origin block number for an Optimism block?

not sure what you want,do you mean you want to know the latest safe block of L2(or its l1origin)?

**Reply by johnhckuo** [reply]
Hi @imtipi , I want to know the oldest l1 origin of a L2 block. It is used at here: https://github.com/ethereum-optimism/optimism/blob/develop/op-batcher/batcher/channel_manager.go#L323

Is there any way to get that from RPC endpoints? 
or any indirect way to calculate this L1 block?
thanks!

**Reply by opfocus** [reply]
https://github.com/ethereum-optimism/optimism/blob/develop/op-batcher/batcher/channel_builder.go#L144    ` // OldestL1Origin returns the oldest L1 block origin from all the L2 blocks that have been added to the channel`

**Reply by johnhckuo** [reply]
I've read the code.
Just asking if there is any api or indirect way we can query/calculate this from outside the node.
